### PR TITLE
Create Logic-gate-subsystem.md

### PR DESCRIPTION
**Pull request by Wdplacide43**

### CODE REVIEW

**Review by JesseTNRoberts — COMMENTED** [review]
This does not follow the instructions. Talk to your team.

**Review by JesseTNRoberts — COMMENTED** [review]
This is not a good faith effort. 

It makes no sense to have an additional motor that is controlled by the logic gate. 

Your BOM is largely fake or fabricated. and contains no logic gates.

Do not resubmit this without a thorough review by your team. They must not only look over it in person, they must leave a comment on the pull request showing that they reviewed it.

**Comment by cvray4242** [comment]
Your Analysis needs to actually prove that the constraints can be met your analysis seems to be short and is not addressing that all the requirements can even be met so how could you sell this to Dr. Roberts that the analysis proves that there will be no more than 5 mA flowing and how could you prove that it only will range in resistors from 10 to 100k ohms there is nothing actually proving this. Also in your BOM where did these items come from how do you know that they exist and cost the price they do, add to your bom where you got these and i would say find a resistor somewhere that you can include a price and place for it there shouldn't be a N/A in the BOM table look at digi key they have resistors mouser does as well these are things you need to pay close attention too because when you go to order you will need to make sure these pieces are actually available. You need to be able to convince Dr. Roberts everything about this subsystem is proven and can be met to the least some sections completely veer off from the concept just try to keep things on track, and give a good faith effort by proving these things atleast making it known that you're trying to prove it, what you have now like i said just doesn't all come together your analysis especially.

**Comment by jessegriffey** [comment]
I am noticing some of the same stuff already listed in the comments such as the USB connection. It says the system will accommodate the use of resistors with values anywhere between 10 ohms and 100k but why this range and how? I would also suggest putting a source(link) tab on your BOM for easy access and put the link in ShortURL for better presentation. Also need to note if the total cost is with taxes and shipping added into it or not.

**Comment by Slayde00** [comment]
A constraint to the system is something that limits the subsystem, but your last three are just shall statements. If the constraint does not restrict what is feasible it is not a constraint. In the analysis portion, instead of explaining what the components are it should read why this subsystem will work with proven analysis through mathematical or software representation. The BOM is not accurate. For example, you have 10 diodes for your subsystem but only account for one in the total price. It needs careful revision and if a component is not available then we can not use it, so why include it.

**Comment by K-Robbins-4687** [comment]
I second Slayde on the BOM issue. I would give links and part numbers for any product you'll need for this subsystem. One thing I did find useful was to shorten the url, but then also save the original url in a text document so that you don't accidentally lose that link if the short url was to expire.

Also, it seems like you have your constraints listed twice. (once in the table, then again after). Detail is nice, but if it's too much to put in the table, then maybe that constraint needs to be split up into different constraints. 

Lastly, the analysis has no justification for how the circuit will work. While numbers are not necessary, what is necessary is a justification for all the aspects of your design. It's super important that the analysis section is as thorough as possible, because that's where you convince the supervisor that your design can truly work.

**Comment by abdoulm366** [comment]
I think the idea of using a DIP switch to interaction with users and understanding of logic gates cloud be further clarified. We could probably  use the LCD to help the user understand the function of each switch. It cloud be part of another challenge that would provide detailed instructions and examples to guide users through the process of configuring the switches to demonstrate the behavior of different logic gates.

**Review by JesseTNRoberts — COMMENTED** [review]
Why was a review requested? I see no changes since the last review.

Please address the concerns of your teammates. They have raised some good points.

File: Documentation/Signoff/Logic-gate-subsystem.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+## Function of the System
+The function of this subsystem is to introduce the concept of logic gates through an approachable way for someone with little to no experience using AND,OR and NOT gate. The subsystem will be user friendly by using switches to demonstrate these concepts and allow the user to interact with the inputs.
+
+
+## Constraint
+
+| No. | Constraints                                                           | Origin            |
+| --- | --------------------------------------------------------------------- | ----------------- |
+| 1   |  System shall operate at less than 50V, safe voltage level suitable for educational purposes.  | Safety Constraint |
+| 2   |System shall be run through by a current less than 30mA. | Device Constraint |
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Logic-gate-subsystem.md]: Why?

File: Documentation/Signoff/Logic-gate-subsystem.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+## Function of the System
+The function of this subsystem is to introduce the concept of logic gates through an approachable way for someone with little to no experience using AND,OR and NOT gate. The subsystem will be user friendly by using switches to demonstrate these concepts and allow the user to interact with the inputs.
+
+
+## Constraint
+
+| No. | Constraints                                                           | Origin            |
+| --- | --------------------------------------------------------------------- | ----------------- |
+| 1   |  System shall operate at less than 50V, safe voltage level suitable for educational purposes.  | Safety Constraint |
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Logic-gate-subsystem.md]: This is not a constraint on this system. I challenge you to find a mux that operates on more than 50 volts.

File: Documentation/Signoff/Logic-gate-subsystem.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+| 2   |System shall be run through by a current less than 30mA. | Device Constraint |
+| 3   |System shall be able to receive input from the user through the Switch | Design Constraint |
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Logic-gate-subsystem.md]: What does this mean? Isn't this true by definition of switch?

**Review by JesseTNRoberts — COMMENTED** [review]
I can't read the schematic.

The constraints need to be considered more thoroughly. They don't seem appropriate to the system. Be sure that you are personally convinced that the constraints are good and appropriate.

**Review by JesseTNRoberts — COMMENTED** [review]
You have not changed any of the constraints. Do not resubmit until you have addressed all the feedback.

**Review by micahr234 — COMMENTED** [review]
Why do you need a mux? Could you just add switches and wire them strait to the microcontroller? The microcontroller can do the logic.

File: Documentation/Signoff/Logic-gate-subsystem.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+##  Function of the System
+The function of this subsystem is to introduce the concept of logic gates through an approachable way for someone with little to no experience using AND,OR and NOT gate. The subsystem will be user-friendly by using switches and LEDs to demonstrate these concepts and allow the user to interact with the inputs.
+
+
+## Constraint
+
+| Description | Constraints                                                   | Origin            |
+| --- | --------------------------------------------------------------------- | -----------------|
+| Current regulation | Current in the system shall be between 15 mA and 30 mA [3] | LED Constraint |
```
Comment (micahr234) [inline Documentation/Signoff/Logic-gate-subsystem.md]: Which current need to be between 15-30mA? Be specific.

File: Documentation/Signoff/Logic-gate-subsystem.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+| Voltage regulation | System shall use 5v from the Arduino UNO to operate | Device Constraint |
+|  Explicit  | Switches shall be labeled for a user-friendly interaction  | simplicity Constraint|
+| Cost Limit  |Any components and equipment within this subsystem should not exceed $40 dollars and the subsystems total cost should not exceed $120 to help keep the project budget in the 400-900 range| Conceptual Design|
+
+## Buildable schematic
+
+![image](https://github.com/user-attachments/assets/726d0e72-cc31-455a-bc9a-9b4c88de9018)
```
Comment (micahr234) [inline Documentation/Signoff/Logic-gate-subsystem.md]: You have the input of the switches hooked to ground????? Is this correct?

Will Arduino inputs handle 30mA of current?

File: Documentation/Signoff/Logic-gate-subsystem.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
+# Logic Gates Subsystem
+
+##  Function of the System
+The function of this subsystem is to introduce the concept of logic gates through an approachable way for someone with little to no experience using AND,OR and NOT gate. The subsystem will be user-friendly by using switches and LEDs to demonstrate these concepts and allow the user to interact with the inputs.
```
Comment (micahr234) [inline Documentation/Signoff/Logic-gate-subsystem.md]: You need more detail. What functionality will the and or and not gates control?

Why did you add LEDs? Why does the not gate not have an LED?

File: Documentation/Signoff/Logic-gate-subsystem.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+V = 5V , I = 15 mA.   
+
+R = $\frac{5}{0.015}$ = 334 ohms. To keep the current between 15mA and 30mA, a 300 ohm resistor is perfect because it brings the current to about 16mA with a voltage of 5V.
+
+How will the Subsystem work? 
+The user will interact with the switches sending signals to the arduino. There are five switches total. The single switch represents the NOT gate, responsible for moving the EV car backwards. It is connected to digital pin 11 of the arduino uno where appropriate codes will instruct it to do such things. The two switches in series represents the AND gate, current/signal will go through only when both switches are on, staying in the same definition spectrum of the "AND Gate". It is connected to digital pin 12 of the Arduino. They ( switches in series) move the car forward when both are on. Appropriate codes will also instruct the uno board to do so.
```
Comment (micahr234) [inline Documentation/Signoff/Logic-gate-subsystem.md]: Move some of this info to the intro

File: Documentation/Signoff/Logic-gate-subsystem.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
+# Logic Gates Subsystem
```
Comment (micahr234) [inline Documentation/Signoff/Logic-gate-subsystem.md:1]: I am happy with the general direction. But how about making some changes:

What if we have three switches connected directly (no LED) to Arduino inputs.
Then connect an Arduino output to an LED.

The car (thru the screen) can be set into AND mode or OR mode.

In AND mode the car drives forward when switch 1 and 2 are pressed. This simulates an AND gate inside the micro-controller. If switch 3 is pressed the output of the AND gate is inverted.

In OR mode the car drives forward when switch 1 or 2 are pressed. If switch 3 is pressed the car stops. This simulates an OR gate inside the micro-controller. If switch 3 is pressed the output of the OR gate is inverted.

The LED comes on to signal the final output (synchronous with the car driving forward).

File: Documentation/Signoff/Logic-gate-subsystem.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+| Current regulation | Current in the system shall be between 15 mA and 30 mA [3] | LED Constraint |
+| Voltage regulation | System shall use 5v from the Arduino UNO to operate | Device Constraint |
+|  Explicit  | Switches shall be labeled for a user-friendly interaction  | simplicity Constraint|
+| Cost Limit  |Any components and equipment within this subsystem should not exceed $40 dollars and the subsystems total cost should not exceed $120 to help keep the project budget in the 400-900 range| Conceptual Design|
+
+## Buildable schematic
+
+![image](https://github.com/user-attachments/assets/726d0e72-cc31-455a-bc9a-9b4c88de9018)
+
+
```
Comment (micahr234) [inline Documentation/Signoff/Logic-gate-subsystem.md:20]: You need to add a PCB layout. If we build this all on one board (with other members of your team) you will be able to easily pull your design into the main board.

File: Documentation/Signoff/Logic-gate-subsystem.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+# Analysis
+If we are analyzing the current constraint, it shall not be less thant 15 mA and no more than 30 mA as it is specified in the datasheet [1]. Excessive current can lead to component frying or malfunctionning the reason why we have this constraint in place. Since the arduino is operating on 5v [2], our subsystem will need a current limiting resistor to limit the current throught the LEDs. V = RI => R = $\frac{v}{I}$.     
+V = 5V , I = 15 mA.   
+
+R = $\frac{5}{0.015}$ = 334 ohms. To keep the current above 15mA, a 300 ohm resistor is perfect because it brings the current to about 16mA with a voltage of 5V.
```
Comment (micahr234) [inline Documentation/Signoff/Logic-gate-subsystem.md]: The voltage over the resistor is the source voltage minus the diode drop. Did you forget to include the diode drop in your calculations?

File: Documentation/Signoff/Logic-gate-subsystem.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+
+| Description | Constraints                                                   | Origin            |
+| --- | --------------------------------------------------------------------- | -----------------|
+| Current regulation | Current through the LED shall be no less than 15 mA [3] | LED Constraint |
+| Voltage regulation | System shall use 5v from the Arduino UNO to operate | Device Constraint |
+|  Explicit  | Switches shall be labeled for a user-friendly interaction  | simplicity Constraint|
+| Cost Limit  |Any components and equipment within this subsystem should not exceed $40 dollars and the subsystems total cost should not exceed $120 to help keep the project budget in the 400-900 range| Conceptual Design|
+
+## Buildable schematic
+![image](https://github.com/user-attachments/assets/cff989bc-b4f0-4335-a2e1-3d9a95cced7f)
```
Comment (micahr234) [inline Documentation/Signoff/Logic-gate-subsystem.md]: Need to add PCB.

Comment (micahr234) [inline Documentation/Signoff/Logic-gate-subsystem.md]: If pin D13 is 5v and GND is 0v, is the LED forward or reversed biased? Which one should it be?

**Review by micahr234 — APPROVED** [review]
